staging/src/k8s.io/apiserver/pkg/storage/cacher.go

```diff
@@ -526,11 +526,17 @@ func (c *Cacher) GuaranteedUpdate(
 	} else if exists {
 		currObj, copyErr := c.copier.Copy(elem.(*storeElement).Object)
 		if copyErr == nil {
-			return c.storage.GuaranteedUpdate(ctx, key, ptrToType, ignoreNotFound, preconditions, tryUpdate, currObj)
+			err := c.storage.GuaranteedUpdate(ctx, key, ptrToType, ignoreNotFound, preconditions, tryUpdate, currObj)
```

Comment:
I'm ok with the structure of this, it's pretty clear what is happening and is lower risk for 1.6.  A TODO  would be good to call out that we may want to consider the alternative approach.

Waiting for Jordan's feedback